### PR TITLE
fix: support SW usage

### PR DESCRIPTION
**Pull request by escapedcat**
We stumbled across [this issue](https://github.com/getAlby/lightning-browser-extension/pull/1124#issuecomment-1476409179) in our PR switching the extension to manifest version 3 (MV3).  

In MV3 the code is running within a service worker so no `window` exitsts. Not exactly sure where the `document` error is coming from but with thsi change it works.

What do you think? Would this work for all use cases?

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -159,9 +159,9 @@ class LnMessage {
     this.connectionStatus$.next('connecting')
     this._attemptReconnect = attemptReconnect
 
-    this.socket = new (
-      typeof window === 'undefined' ? (await import('ws')).default : window.WebSocket
-    )(this.wsUrl)
+    // https://stackoverflow.com/questions/17575790/environment-detection-node-js-or-browser#comment120214255_31456668
+    const isNode = typeof process !== 'undefined' && process?.versions?.node
+    this.socket = new (isNode ? (await import('ws')).default : WebSocket)(this.wsUrl)
```
Comment (bumi) [inline src/index.ts]: I think this should use `globalThis` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis

maybe:
```suggestion
    this.socket =  new (typeof globalThis.WebSocket === 'undefined' ? (await import('ws')).default : globalThis.WebSocket)(this.wsUrl)
```

**Comment by lnbc1QWFyb24** [comment]
Yeah `globalThis` seems like the way to go. Are you going to update to use that instead?

**Comment by escapedcat** [comment]
Done

**Review by lnbc1QWFyb24 — APPROVED** [review]
Awesome, looks good!

**Comment by lnbc1QWFyb24** [comment]
Hey @escapedcat I have published a [beta module](https://www.npmjs.com/package/lnmessage/v/0.1.0-0.3.0) that you can test before I release fully. Could you let me know if that looks all good on your end?

**Comment by escapedcat** [comment]
Funny version :) Works! Thanks!

**Pull request merged**